core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -239,9 +239,10 @@ protected void doPost(HttpServletRequest httpServletRequest,
             {
             	String extraMessage = "";
             	//extra message for the cases where property is missing (will happen often in transition period to this new versioning model):
-            	if (dbPortalVersion.equals("0"))
+            	if (dbPortalVersion.equals("0")) {
             		extraMessage = "The db.version property also not found in your portal.properties file. This new property needs to be added by the administrator.";
-                httpServletRequest.setAttribute(DB_ERROR, "Current DB Version: " + dbVersion + "<br/>" + "DB version expected by Portal: " + dbPortalVersion + "<br/>" + extraMessage);
+            	}
+                throw new DbVersionException("Current DB Version: " + dbVersion + "<br/>" + "DB version expected by Portal: " + dbPortalVersion + "<br/>" + extraMessage);
```

Comment:
It seems a little odd to throw and catch an exception within a single function call, but I suppose it is better than a "goto" style statement. It seems like the block of catch statements below create a priority order however .. where only the first applicable catch phrase will be executed, and other (possibly detectable) conditions are not reported. Since that seems to be the case, it would be equivalent to just inline the catch block in place of the throw statement. Then a new custom exception class would not be needed.